File: EduMS-Frontend/edums-frontend-DDD-System/src/app/modules/m1-school-office/interface/school-announcement-log.ts
import type { TargetAudience, Priority } from './common';

export interface SchoolAnnouncementLog {
  id: number;
  schoolId: number;
  titleAr: string;
  titleEn: string;
  announcementContent: string;
  publishDate: string;
  expireDate: string | null;
  targetAudience: TargetAudience;
  isPinned: boolean;
  announcementPriority: Priority;
  attachmentFileUrl: string | null;
  viewCount: number;
  publishedByEmployeeId: number | null;
  isActive: boolean;
}

export type CreateSchoolAnnouncementLogDto = Omit<SchoolAnnouncementLog, 'id' | 'viewCount' | 'isActive'>;

export type UpdateSchoolAnnouncementLogDto = Omit<SchoolAnnouncementLog, 'schoolId' | 'viewCount' | 'isActive'>;

File: 01_Backend/EduMS.Backend/src/EduMS.Domain/Entities/M1_SchoolAdmin/SchoolAnnouncementLog.cs
using EduMS.Domain.Common;

namespace EduMS.Domain.Entities;

public class SchoolAnnouncementLog : BaseAuditableEntity
{
    public long SchoolId { get; set; }
    public string TitleAr { get; set; } = string.Empty;
    public string TitleEn { get; set; } = string.Empty;
    public string AnnouncementContent { get; set; } = string.Empty;
    public DateTime PublishDate { get; set; } = DateTime.UtcNow;
    public DateTime? ExpireDate { get; set; }
    public int TargetAudience { get; set; } // 1=Students, 2=Guardians, 3=Teachers, 4=All
    public bool IsPinned { get; set; } = false;
    public int AnnouncementPriority { get; set; } = 2; // 1=Low, 2=Normal, 3=High, 4=Urgent
    public string? AttachmentFileUrl { get; set; }
    public int ViewCount { get; set; }
    public long? PublishedByEmployeeId { get; set; }
    public bool IsActive { get; set; } = true;

    // Navigation Property
    public virtual School? School { get; set; }
    public virtual Employee? PublishedByEmployee { get; set; }
}
